src/deepsparse/loggers/metric_functions/cv/built_ins.py

```diff
@@ -0,0 +1,273 @@
+
+     :param classes: A nested list, where:
+        - first level is the batch information
+        - second level is the sample information. Can be
+            either `None` (no detection) or an integer/string
+            representation of the class label
+    :return: Dictionary, where the keys are image indices within
+        a batch and the values are the number of detected objects per image
+    """
+    _check_valid_classes(classes)
+
+    number_objects_per_batch = {}
+    for sample_idx, sample in enumerate(classes):
+        number_objects_per_batch[str(sample_idx)] = (
+            0 if sample == [None] else len(sample)
+        )
+
+    return number_objects_per_batch
+
+
+def mean_score_per_detection(scores: List[List[Optional[float]]]) -> Dict[str, float]:
+    """
+    Return the mean score per detection
+
+    :param scores: A nested list, where:
+        - first level is the batch information
+        - second level is the sample information. Can be
+            either `None` (no detection) or a float score
+    :return: Dictionary, where the keys are image indices within
+        a batch and the values are the mean score per detection
+    """
+    _check_valid_scores(scores)
+
+    mean_scores_per_batch = {}
+    for sample_idx, sample in enumerate(scores):
+        if sample == [None]:
+            mean_scores_per_batch[str(sample_idx)] = 0.0
+        else:
+            mean_scores_per_batch[str(sample_idx)] = numpy.mean(sample)
+
+    return mean_scores_per_batch
+
+
+def std_score_per_detection(scores: List[List[Optional[float]]]) -> Dict[str, float]:
+    """
+    Return the standard deviation of scores per detection
+
+    :param scores: A nested list, where:
+        - first level is the batch information
+        - second level is the sample information. Can be
+            either `None` (no detection) or a float score
+    :return: Dictionary, where the keys are image indices within
+        a batch and the values are the standard deviation of scores per detection
+    """
+    _check_valid_scores(scores)
+
+    std_scores_per_batch = {}
+    for sample_idx, sample in enumerate(scores):
+        if sample == [None]:
+            std_scores_per_batch[str(sample_idx)] = 0.0
+        else:
+            std_scores_per_batch[str(sample_idx)] = numpy.std(sample)
+
+    return std_scores_per_batch
+
+
+def _check_valid_classes(classes: List[List[Optional[Union[int, str]]]]):
+    for sample in classes:
+        if (
+            all(isinstance(i, int) for i in sample)
+            or all(isinstance(i, str) for i in sample)
+            or sample == [None]
+        ):
+            pass
+        else:
+            raise ValueError(
+                "Detection for a sample must be either a "
+                "list of integers or a list of strings or a list "
+                "with a single `None` value"
+            )
+
+
+def _check_valid_scores(scores: List[List[Optional[float]]]):
```

Comment:
what's the intention of adding these checks? especially when called multiple times on the built in functions for the same outputs called directly from pipelines who are designed to have these exact output types